bindsnet/network/__init__.py

```diff
@@ -258,6 +266,15 @@ def run(self, inpts: Dict[str, torch.Tensor], time: int, **kwargs) -> None:
         masks = kwargs.get('masks', {})
         injects_v = kwargs.get('injects_v', {})
 
+        # Compute reward prediction error.
+        if self.reward_fn is not None:
```

Comment:
I think this is good for your use case, but we might want something more general in the future. We might want `reward_fn` to compute a reward conditioned on the output of a layer, for example.

Reply:
To that end, we wouldn't require a `reward` argument, but your `MovingAverageRPE` does, which is where you could move the `try / except` block.